src/backend/cuda/kernel/iota.hpp

```diff
@@ -41,15 +41,15 @@ __global__ void iota_kernel(Param<T> out, const int s0, const int s1,
 
     const int ozw = ow * out.strides[3] + oz * out.strides[2];
 
-    T val = (ow % s3) * s2 * s1 * s0;
+    dim_t val = (ow % s3) * s2 * s1 * s0;
     val += (oz % s2) * s1 * s0;
 
     const int incy = blocksPerMatY * blockDim.y;
     const int incx = blocksPerMatX * blockDim.x;
 
     for (int oy = yy; oy < out.dims[1]; oy += incy) {
         int oyzw = ozw + oy * out.strides[1];
-        T valY   = val + (oy % s1) * s0;
+        dim_t valY   = val + (oy % s1) * s0;
```

Comment:
Same as CPU, it is odd that we used to stored indexing type in `T` which is most likely template parameter.